src/libbson/src/bson/bson.c

```diff
@@ -3584,7 +3584,15 @@ _bson_iter_validate_document (const bson_iter_t *iter,
       state->phase = BSON_VALIDATE_PHASE_LF_REF_KEY;
    }
 
-   (void) bson_iter_visit_all (&child, &bson_validate_funcs, state);
+   bson_iter_visit_all_v2 (
+      &child, &bson_validate_funcs, BSON_ITER_VISIT_VALIDATE_KEYS, state);
```

Comment:
If `state.flags` has `BSON_VALIDATE_UTF8`, should `BSON_ITER_VISIT_VALIDATE_STRINGS` be passed instead?

Same question for `BSON_ITER_VISIT_ALLOW_NUL_IN_UTF8`.